File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/web/mappers/SurveysDTOsMapper.java
package io.zenwave360.example.clinicaltool.modules.surveys.web.mappers;

import io.zenwave360.example.clinicaltool.common.mappers.*;
import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*;
import io.zenwave360.example.clinicaltool.modules.surveys.web.dtos.*;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;
import java.math.*;
import java.time.*;
import java.util.*;
import org.springframework.data.domain.Page;

@Mapper(uses = BaseMapper.class)
public interface SurveysDTOsMapper {

    SurveysDTOsMapper INSTANCE = Mappers.getMapper(SurveysDTOsMapper.class);

    // request mappings
        SurveyAnswers asSurveyAnswers(SurveyAnswersDTO dto);

    // response mappings
    
    SurveyAnswersDTO asSurveyAnswersDTO(SurveyAnswers entity);
    
    
    SurveyAndQuestionsDTO asSurveyAndQuestionsDTO(SurveyAndQuestions entity);
    


}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/example/asyncapi/provider/clinicaltool/modules/surveys/web/mappers/SurveysDTOsMapper.kt
package io.example.asyncapi.provider.clinicaltool.modules.surveys.web.mappers

import io.zenwave360.example.clinicaltool.common.mappers.*
import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*
import io.zenwave360.example.clinicaltool.modules.surveys.web.dtos.*

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.factory.Mappers
import java.math.*
import java.time.*
import java.util.*
import org.springframework.data.domain.Page

@Mapper(uses = [BaseMapper::class])
interface SurveysDTOsMapper {

    companion object {
        val INSTANCE: SurveysDTOsMapper = Mappers.getMapper(SurveysDTOsMapper::class.java)
    }

    // request mappings
    fun asSurveyAnswers(dto: SurveyAnswersDTO): SurveyAnswers

    // response mappings

    fun asSurveyAnswersDTO(entity: SurveyAnswers): SurveyAnswersDTO


    fun asSurveyAndQuestionsDTO(entity: SurveyAndQuestions): SurveyAndQuestionsDTO


}
